fix(shared-data): Sync labware dimensions and spacing to latest hardware values

SyntaxColoring opened the pull request:
## Overview

Closes EXEC-1480.

## Changelog

* Copies the numbers from the "XY Spacing" tab of the Google Sheet linked in EXEC-1480.

  This means updating the coarse-grained labware and well dimensions, but *not* the fine-grained `innerWellGeometry`. `innerWellGeometry` should have already been sorted out by #18266.
  * This was partially [script-assisted](https://github.com/SyntaxColoring/labware-scripts/blob/main/import_dimensions_from_hw.py), but only partially. The details are in the commit messages.
  
  * **Unfortunately, I couldn't bring the tube racks fully up to date** because the spreadsheet is missing some details. It tells us tube depths, but it doesn't tell us anything about where the tube sits vertically. So if we need to fix the depth, we have no way of knowing whether to grow/shrink it from the top or the bottom.

* Adjust our labware definition tests to only check each labware's *latest version* for sensible geometry.

## Test Plan and Hands on Testing

Relying on code review and automated labware definition sanity checks to catch any flagrant errors.

## Review requests

Basically just double-check my new numbers against the numbers in the Google Sheet and make sure I didn't make any silly mistakes like copying from the wrong place.

There is one tricky thing related to well x- and y-offset. The Google Sheet seems to treat the x- and y-offset as a measurement from the labware back-left corner (-x, +y) to the center of the back-left well. (I'm inferring this by looking at the errors described in the Sheet, and because it's the standard way of doing dimensions in technical drawings, as I understand it.) This means, if you're skimming the JSON files, the well `x` and `y` will unfortunately often *not* literally match what's written out in the Sheet.

## Risk assessment

Low. Most of the changes are pretty small, numerically.

SyntaxColoring committed:
Sync dimensions, spacing, diameter, and depth from the hardware spreadsheet, via script.

Notes:

- Run import_dimensions_from_hw.py.
- Manually fix up the output to reduce the diff. e.g. undo changes like 0->0.0.
- Opentrons tube rack labware like opentrons_24_tuberack_generic_2ml_screwcap are incomplete in the spreadsheet and couldn't be parsed by the script. They need separate manual attention.

SyntaxColoring committed:
Test xDimension/yDimension/diameter, in addition to depth, against geometry.

SyntaxColoring committed:
Manually fix nest_1_reservoir_195ml topXDimension/topYDimension.

x/y were swapped in innerLabwareGeometry.

SyntaxColoring committed:
Limit geometry tests to just the most recent version of each labware.

SyntaxColoring committed:
Create drafts from the latest versions of each labware.

SyntaxColoring committed:
Manually fix opentrons_tough_1_reservoir_300ml xDimension/yDimension.

New numbers based on the drawing "SINGLE WELL RESERVOIR SUPPORT DWG" rev A1.0.

xDimension 108->108.7: The old value matched the bottom of the well, but a well's xDimension needs to represent the top of the well. 108.7 matches the drawing and the number in innerLabwareGeometry.

yDimension 72->70.7: I'm not sure where 72 came from, but 70.7 matches the drawing and the number in innerLabwareGeometry.

SyntaxColoring committed:
Manually fix aluminum block tube diameters.

These "aluminum block + tube" labware are not mentioned in the hardware spreadsheet. But, they have the same tubes as some "tube rack + tube" labware, which means we can take the diameters from there.

* opentrons_24_tuberack_nest_0.5ml_screwcap <- opentrons_24_tuberack_nest_0.5ml_screwcap
* opentrons_24_aluminumblock_nest_1.5ml_screwcap <- opentrons_24_tuberack_nest_1.5ml_screwcap
* opentrons_24_aluminumblock_nest_1.5ml_snapcap <- opentrons_24_tuberack_nest_1.5ml_snapcap
* opentrons_24_aluminumblock_nest_2ml_screwcap <- opentrons_24_tuberack_nest_2ml_screwcap
* opentrons_24_aluminumblock_nest_2ml_snapcap <- opentrons_24_tuberack_nest_2ml_snapcap

This also makes each definition's tube diameter internally consistent with the diameter implied by its innerLabwareGeometry, yay.

SyntaxColoring committed:
Manually sync tube rack diameters from the hardware spreadsheet.

We can't easily use a script for this because the spreadsheet doesn't have full information for these definitions--it only has the diameters and depths. Also, some of the tube racks are heterogeneous.

Also, we can only copy over the diameter values, not the depth values. Since the hardware spreadsheet doesn't give us a known tube top position, we can't know whether to grow/shrink the tube depths from the bottom or the top.

SyntaxColoring committed:
Remove test xfails for labware that we've just fixed.

SyntaxColoring committed:
Fix diameter in opentrons_6_tuberack_nest_50ml_conical.

The hardware spreadsheet doesn't explicitly mention the diameter of these tubes, so this is a little bit dicey.

These new numbers make the well diameter internally consistent with the definition's innerLabwareGeometry.

A while ago, in PR #17082, this same 27.95 -> 28.18 change was made to opentrons_10_tuberack_nest_4x50ml_6x15ml_conical, which shares the same tubes. I'm guessing this definition was just accidentally omitted.

SyntaxColoring committed:
Delete drafts that we didn't change.

SyntaxColoring committed:
Copy drafts that we changed to new labware versions.

SyntaxColoring committed:
Add xfails for new labware.

I've asked @rclarke0 and @HovanNgoOpentrons about these. They'll probably get fixed in a separate PR.

SyntaxColoring commented on shared-data/js/__tests__/labwareDefSchemaV2.test.ts:
```diff
@@ -320,59 +294,63 @@ const checkGeometryDefinitions = (labwareDef: LabwareDefinition2): void => {
       } else {
-        expect(wellDepth).toStrictEqual(topFrustumHeight)
+        expect(xyDimensionsFromWell).toStrictEqual(xyDimensionsFromGeometry)
       }
```
@rclarke0 FYI here's where I ended up with this test. It's currently missing the handling that you wrote for circular sections, but if you end up keeping that, I'll merge it in here somehow.

SyntaxColoring committed:
nest_1_reservoir_195ml: Complete my incomplete fix for swapped xDimension/yDimension.

SyntaxColoring committed:
Update Quick Transfer constants.

rclarke0 commented on shared-data/labware/definitions/2/agilent_1_reservoir_290ml/4.json:
```diff
@@ -0,0 +1,80 @@
+      "yDimension": 71.25,
+      "totalLiquidVolume": 290000,
+      "x": 63.88,
+      "y": 42.74,
```
the google sheet says 42.73

SyntaxColoring replied:
Probably the same thing as https://github.com/Opentrons/opentrons/pull/19127#discussion_r2259067994.

rclarke0 commented on shared-data/labware/definitions/2/corning_24_wellplate_3.4ml_flat/5.json:
```diff
@@ -0,0 +1,342 @@
+      "diameter": 16.256,
+      "totalLiquidVolume": 3400,
+      "x": 113.96,
+      "y": 13.79,
```
the sheet says the y offset is 13.78 so I think this should also be 13.78 since its the bottom most well

SyntaxColoring replied:
Probably the same thing as https://github.com/Opentrons/opentrons/pull/19127#discussion_r2259067994.

rclarke0 commented on shared-data/labware/definitions/2/corning_384_wellplate_112ul_flat/5.json:
```diff
@@ -0,0 +1,5116 @@
+      "yDimension": 3.632,
+      "totalLiquidVolume": 112,
+      "x": 115.62,
+      "y": 8.98,
```
the sheet says the Y offset is 8.99

SyntaxColoring replied:
Probably the same thing as https://github.com/Opentrons/opentrons/pull/19127#discussion_r2259067994.

rclarke0 commented on shared-data/labware/definitions/2/corning_48_wellplate_1.6ml_flat/6.json:
```diff
@@ -0,0 +1,610 @@
+      "diameter": 11.1,
+      "totalLiquidVolume": 1600,
+      "x": 109.72,
+      "y": 10.12,
```
y offset is 10.08 on the sheet

SyntaxColoring replied:
This is interesting, thanks for calling this out.

According to the spreadsheet:

* y-spacing: 13.08 
* y-offset: 10.08
* y-dimension of overall labware: 85.600

If you treat y-offset as the distance from the front of the labware to the center of the "F" row, then this `y` value should be 10.08. This is what you're doing.

If you treat y-offset as the offset from the back of the labware to the center of the "A" row, then this `y` value should be `85.600 - 10.08 - 13.08*5 = 10.12`. This is what I did. I did this because, as I understand it, the standard way of dimensioning things like this is from the back-left. 

I have to imagine that, physically, Corning intends for the well distribution to be exactly symmetrical, evenly dividing the space across 6 rows. Somewhere along the line—maybe Corning's technical drawings, maybe our spreadsheet—the y-spacing or y-offset is getting rounded, and the space is no longer divided exactly evenly.

So our options are:

1. Make the y-offset match the spreadsheet in the A-row but not the F-row. (This is what this PR has now.)
2. Make the y-offset match the spreadsheet in the F-row but not the A-row.
3. Make the y-offset match the spreadsheet in the A-row *and* the F-row, but then the y-*spacing* doesn't match the spreadsheet anymore.

My initial reaction is that I'm 50/50 between option 1 and option 2. I don't think we should do option 3.

SyntaxColoring replied:
OK, after looking at the spreadsheet again, I'm leaning towards option 1, because I think that matches how the spreadsheet is thinking about it. In other words, I'm pretty sure the spreadsheet is thinking of y-offset as a measurement from the back of the labware to the center of the "A" row. Notice how in the "ERROR" section, this labware had an error of -0.040, which is corrected here.

rclarke0 replied:
Makes sense, I agree with option 1. I think this formula should get added to the PR description so that its clear why the offset numbers in the sheet aren't directly present in the labware definition. I guess this is still a result of the issue that the labware does not assume the top left corner is (0,0).

rclarke0 commented on shared-data/labware/definitions/2/corning_6_wellplate_16.8ml_flat/5.json:
```diff
@@ -0,0 +1,134 @@
+      "diameter": 35.433,
+      "totalLiquidVolume": 16800,
+      "x": 103,
+      "y": 23.19,
```
y offset is 23.16

SyntaxColoring replied:
Probably the same thing as https://github.com/Opentrons/opentrons/pull/19127#discussion_r2259067994.

rclarke0 commented on shared-data/labware/definitions/2/corning_96_wellplate_360ul_flat/5.json:
```diff
@@ -0,0 +1,1177 @@
+      "diameter": 6.858,
+      "totalLiquidVolume": 360,
+      "x": 113.37,
+      "y": 11.24,
```
y offset is 11.23

SyntaxColoring replied:
Probably the same thing as https://github.com/Opentrons/opentrons/pull/19127#discussion_r2259067994.

rclarke0 commented on shared-data/labware/definitions/2/nest_1_reservoir_290ml/4.json:
```diff
@@ -0,0 +1,80 @@
+      "yDimension": 71.1,
+      "totalLiquidVolume": 290000,
+      "x": 63.88,
+      "y": 42.73,
```
y offset is 42.74

SyntaxColoring replied:
Probably the same thing as https://github.com/Opentrons/opentrons/pull/19127#discussion_r2259067994.

rclarke0 commented on shared-data/labware/definitions/2/opentrons_10_tuberack_falcon_4x50ml_6x15ml_conical/3.json:
```diff
@@ -0,0 +1,232 @@
+      "totalLiquidVolume": 15000,
+      "diameter": 14.7,
+      "shape": "circular",
+      "depth": 117.5,
```
sheet says 15mL depth should be 118.2

SyntaxColoring replied:
Thanks for noticing, yeah. Unfortunately, I don't think there's anything that I can do about this right now. From the PR description:

> * **Unfortunately, I couldn't bring the tube racks fully up to date** because the spreadsheet is missing some details. It tells us tube depths, but it doesn't tell us anything about where the tube sits vertically. So if we need to fix the depth, we have no way of knowing whether to grow/shrink it from the top or the bottom.

rclarke0 commented on shared-data/labware/definitions/2/opentrons_15_tuberack_falcon_15ml_conical/3.json:
```diff
@@ -0,0 +1,259 @@
+  "namespace": "opentrons",
+  "wells": {
+    "C1": {
+      "depth": 117.5,
```
sheet says depth is 118.2

SyntaxColoring replied:
Tube rack, same problem as https://github.com/Opentrons/opentrons/pull/19127#discussion_r2259024883.

rclarke0 commented on shared-data/labware/definitions/2/opentrons_24_aluminumblock_nest_1.5ml_screwcap/3.json:
```diff
@@ -0,0 +1,356 @@
+  },
+  "wells": {
+    "A1": {
+      "depth": 43.9,
```
sheet says the depth is 42.6

SyntaxColoring replied:
I'll double-check this, but I suspect these "aluminum block + tube" labwares have the same problem as the "tube rack + tube" labwares where the spreadsheet doesn't give us enough information to know whether the depth should be grown/shrunk from the top or the bottom.

[Edit] Yep, double-checked, not spelled out in the spreadsheet.

rclarke0 commented on shared-data/labware/definitions/2/opentrons_24_aluminumblock_nest_2ml_screwcap/3.json:
```diff
@@ -0,0 +1,350 @@
+  },
+  "wells": {
+    "A1": {
+      "depth": 44.05,
```
depth on sheet is 43.4

SyntaxColoring replied:
Aluminum block, see https://github.com/Opentrons/opentrons/pull/19127#discussion_r2258413826.

rclarke0 commented on shared-data/labware/definitions/2/opentrons_24_tuberack_eppendorf_2ml_safelock_snapcap/3.json:
```diff
@@ -0,0 +1,473 @@
+  },
+  "wells": {
+    "D1": {
+      "depth": 39.1,
```
sheet says depth is 39.02

SyntaxColoring replied:
Tube rack, same problem as https://github.com/Opentrons/opentrons/pull/19127#discussion_r2259024883.

rclarke0 commented on shared-data/labware/definitions/2/opentrons_6_tuberack_nest_50ml_conical/3.json:
```diff
@@ -0,0 +1,150 @@
+  },
+  "wells": {
+    "A1": {
+      "depth": 113.05,
```
depth on sheet is 113.1

SyntaxColoring replied:
Tube rack, same problem as https://github.com/Opentrons/opentrons/pull/19127#discussion_r2259024883.

rclarke0 commented on shared-data/labware/definitions/2/opentrons_tough_1_reservoir_300ml/2.json:
I dont think this needs a 2.json file because it hasn't been released to the public yet

SyntaxColoring replied:
This just happened today, but #19142, which I think is going into v8.6.0, does release it to the public.

(And from speaking with Seth, *this* PR will *not* go into v8.6.0, for risk mitigation reasons. So we gotta leave v1 of this labware alone.)

SyntaxColoring replied:
Reminder to self: This is getting fixed in-place in #19126 so it should probably be removed from this PR.

[Edit] Done.

rclarke0 commented on shared-data/labware/definitions/2/usascientific_12_reservoir_22ml/4.json:
```diff
@@ -0,0 +1,237 @@
+      "yDimension": 71.88,
+      "totalLiquidVolume": 22000,
+      "x": 14.38,
+      "y": 42.71,
```
y offset in sheet is 42.76

SyntaxColoring replied:
Probably the same thing as https://github.com/Opentrons/opentrons/pull/19127#discussion_r2259067994.

rclarke0 commented on shared-data/labware/definitions/2/usascientific_96_wellplate_2.4ml_deep/4.json:
```diff
@@ -0,0 +1,1243 @@
+      "yDimension": 8.2,
+      "totalLiquidVolume": 2400,
+      "x": 14.4,
+      "y": 11.3,
```
y offset on the sheet is 11.2

SyntaxColoring replied:
Probably the same thing as https://github.com/Opentrons/opentrons/pull/19127#discussion_r2259067994.

rclarke0 reviewed (changes requested):
Left comments on files where I saw discrepancies

rclarke0 reviewed (approved):
Thanks for explaining the offsets

SyntaxColoring commented:
Tagging "do not merge": we want to wait until after the v8.6.0 release branch is cut from `edge`.

SyntaxColoring committed:
Merge branch 'edge' into sync_dimensions_and_spacing_from_hw_2

SyntaxColoring committed:
Merge fixup: Delete now-unnecessary v2 of opentrons_tough_1_reservoir_300ml.

SyntaxColoring committed:
Merge fixup: Delete test xfails for eppendorf_96_wellplate_1000ul and nest_8_reservoir_22ml.

These were resolved in edge.

SyntaxColoring committed:
Merge fixup: Merge Rhyanne's new test with mine.

SyntaxColoring committed:
Add labware versions introduced in this PR to apiLevel 2.26.

SyntaxColoring committed:
Also add opentrons_96_aluminumblock_generic_pcr_strip_200ul v4 to apiLevel 2.26.

This was added in a prior PR but never added to an apiLevel, under the assumption that it would be continued to be worked on in this PR, probably to fix up well depths. It didn't actually get changed in this PR because the hardware spreadsheet doesn't give us enough info to know whether the well depths ought to be grown/shrunk from the top or bottom.
